source/agora/consensus/protocol/Nominator.d

```diff
@@ -673,28 +673,34 @@ extern(D):
     public override Value combineCandidates (uint64_t slot_idx,
         ref const(set!Value) candidates)
     {
-        scope (failure) assert(0);
-
-        foreach (ref const(Value) candidate; candidates)
+        try
         {
-            auto data = deserializeFull!ConsensusData(candidate[]);
-
-            if (auto msg = this.ledger.validateConsensusData(data))
+            foreach (ref const(Value) candidate; candidates)
             {
-                log.error("combineCandidates(): Invalid consensus data: {}", msg);
-                continue;
+                auto data = deserializeFull!ConsensusData(candidate[]);
+
+                if (auto msg = this.ledger.validateConsensusData(data))
+                {
+                    log.error("combineCandidates(): Invalid consensus data: {}", msg);
+                    continue;
+                }
+                else
+                {
+                    log.info("combineCandidates: {}", slot_idx);
+                }
+
+                // todo: currently we just pick the first of the candidate values,
+                // but we should ideally pick tx's out of the combined set
+                return duplicate_value(&candidate);
             }
-            else
-            {
-                log.info("combineCandidates: {}", slot_idx);
-            }
-
-            // todo: currently we just pick the first of the candidate values,
-            // but we should ideally pick tx's out of the combined set
-            return duplicate_value(&candidate);
+        } catch (Exception ex)
+        {
+            assert(0, format!"[%s]:[%s] combineCandidates: slot %u. Exception:%s"(
```

Comment:
```suggestion
            assert(0, format!"[%s:%s] combineCandidates: slot %u. Exception: %s"(
```

Using `file:line` ensures that it will be clickable by most IDE.